GliaWidgets/ViewModel/EngagementViewModel.swift

```diff
@@ -128,9 +132,13 @@ class EngagementViewModel {
                 )
             )
         case .ended:
-            if EngagementViewModel.alertPresenters.isEmpty {
-                engagementDelegate?(.finished)
-            }
+            EngagementViewModel.alertPresenters.insert(self)
+            engagementAction?(
+                .showSingleActionAlert(
+                    alertConfiguration.operatorEndedEngagement,
+                    confirmed: { self.endSession() }
```

Comment:
Looks like `[weak self]` is required. Let's use weak reference here to avoid retain cycle.